eng/pipelines/build-windows-job.yml

```diff
@@ -34,6 +34,9 @@ jobs:
   timeoutInMinutes: 40
   variables:
     artifactName: ${{ parameters.testArtifactName }}
+    ROSLYNCOMMANDLINELOGFILE: '$(Build.SourcesDirectory)/artifacts/log/${{ parameters.configuration }}/Build.Server.log'
```

Comment:
How does the correctness job get these env vars set?  From what I can tell it doesn't use the build template here - https://github.com/dotnet/roslyn/blob/main/azure-pipelines.yml#L275

Reply:
Sigh ... it's never easy 😦

Reply:
@dibarbet okay, refactored this a bit. I added a new template which inserts thevariables and put the template into all the jobs I could find.